table.go

```diff
@@ -69,17 +65,25 @@ func NumDigits(b *big.Int) int64 {
 	if bl == 0 {
 		return 1
 	}
-	if val, ok := lookupBits(bl); ok {
-		ab := new(big.Int).Abs(b)
-		if ab.Cmp(&val.border) < 0 {
+
+	if bl < digitsTableLen {
+		val := digitsLookupTable[bl]
+		a := b
+		if b.Sign() < 0 {
+			a = new(big.Int).Abs(b)
+		}
+		if a.Cmp(&val.border) < 0 {
 			return val.digits
 		}
 		return val.digits + 1
 	}
 
 	n := int64(float64(bl) / digitsToBitsRatio)
-	a := new(big.Int).Abs(b)
-	e := new(big.Int).Exp(bigTen, big.NewInt(n), nil)
+	e := tableExp10(n, nil)
+	a := b
+	if b.Sign() < 0 {
+		a = new(big.Int).Abs(b)
```

Comment:
Ah ok, I guess I was confused that the comment talks about what it returns, then talks about what it will set `f` to, then talks about what it returns again. Why not just call `f` `tmpInt` instead and make no guarantee about what it will be set to. This would be a lot more clear, and I doubt we rely on the behavior of `f` being set to `x` anywhere.